src/shapes/text.class.js

```diff
@@ -1515,6 +1517,8 @@
         'charSpacing',
         'path',
         'direction',
+        'pathStartOffset',
+        'pathSide'
       ].concat(propertiesToInclude);
```

Comment:
i wonder if we can replace this custom array here with additionalProps higher up in the file.
Can you check if they match?

Reply:
This file has 3 very similar property arrays (`additionalProperties`, `additionalProps`, and `_dimensionAffectingProps`) . All 3 share the following props:

path, fontSize, charSpacing, fontStyle, fontWeight, pathStartOffset, pathSide, lineHeight, fontFamily, textAlign, text

Here are the props that are not shared by every array:

**additionalProps:** 
textBackgroundColor, linethrough, overline, underline, styles

**additionalProperties (toObject method):** 
textBackgroundColor, linethrough, overline, underline, direction

**_dimensionAffectingProps:** 
styles

Perhaps we could simplify all three arrays by having a master array for the shared values and then doing a concat for the unique values?

Reply:
`direction has to be added to additionalProps`, for sure. That invalidate caching between the things

Reply:
`styles` would be extra in the internal toObject array, but is overridden a line after.

Please add direction to additionalProps, and delete additionalProperties, use additionalProp in its place.